src/legacy/core_plugins/data/public/expressions/expression_runner.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Ast, fromExpression } from '@kbn/interpreter/common';
+
+import { RequestAdapter, DataAdapter } from 'ui/inspector/adapters';
+import { RenderFunctionsRegistry, Interpreter, Result } from './expressions_service';
+
+export type ExpressionRunner = (
+  expressionOrAst: string | Ast,
+  element?: Element
+) => Promise<Result>;
+
+export const createRunFn = (
+  renderersRegistry: RenderFunctionsRegistry,
+  interpreterPromise: Promise<Interpreter>
+): ExpressionRunner => async (expressionOrAst, element) => {
+  const interpreter = await interpreterPromise;
+  const ast =
+    typeof expressionOrAst === 'string' ? fromExpression(expressionOrAst) : expressionOrAst;
+  const response = await interpreter.interpretAst(
+    ast,
+    { type: 'null' },
```

Comment:
context should be one of the properties, needs to be provided from outside and we should default to null.